src/main/java/io/vertx/ext/httpservicefactory/HttpServiceFactory.java

```diff
@@ -0,0 +1,388 @@
+package io.vertx.ext.httpservicefactory;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.file.AsyncFile;
+import io.vertx.core.file.OpenOptions;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.streams.Pump;
+import io.vertx.core.streams.WriteStream;
+import io.vertx.service.ServiceVerticleFactory;
+import org.bouncycastle.openpgp.PGPPublicKey;
+import org.bouncycastle.openpgp.PGPSignature;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.net.URLEncoder;
+import java.nio.file.Files;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.function.Function;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+/**
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ */
+public class HttpServiceFactory extends ServiceVerticleFactory {
+
```

Comment:
Does it support proxies?

There is code that can be reused from Vert.x 2 there too :)